src/main/java/de/flo56958/minetinker/listeners/AnvilListener.java

```diff
@@ -64,11 +64,15 @@ public void onInventoryClick(@NotNull final InventoryClickEvent event) {
 			return;
 		}
 
-		if (!modManager.isModifierItem(modifier)) { //upgrade
-			if (tool.getType().equals(newTool.getType())) return; //Not an upgrade
+		if (!modManager.isModifierItem(item2)) { //Upgrade or combining
+			if (item1.getType().equals(newTool.getType())) { //Combining
+				player.setItemOnCursor(newTool);
```

Comment:
This would break with Shiftclicking the Item. Youll also need to check for shift clicking. (see below)